### ISSUE TITLE
dnspython dependency version overly restrictive

### ISSUE THREAD

**Issue by moorecode** [nameko/nameko#760]
**Describe the bug**
[This issue](https://github.com/rthalley/dnspython/issues/559) indicates that `eventlet` versions < `0.26.0` are incompatible with `dnspython >= 2`

Is it possible to remove this requirement in nameko for python 3.8 with `eventlet > 0.26.0`?

**Environment:**
 - Nameko version: 2.13.0
 - Python version: 3.8.12
 - Eventlet version: 0.33.0
 - OS: Linux

**Comment by mattbennett** [comment]
I think I am in favour of removing the `dnspython` pin from nameko completely now that modern releases of eventlet are compatible.

Nameko could still allow older versions of eventlet to be used if necessary, but library authors will have to add their own pins for dnspython in that case. I think this would be better than either upping the minimum eventlet version to 0.26.0 or keeping this overly restrictive pin for any versions of Python.

@moorecode what do you think?

**Comment by moorecode** [comment]
Makes sense to me, I'm in favor of allowing other devs to control this rather than pin it in nameko

**Comment by fechnert** [comment]
I've opened a pull request for this. See https://github.com/nameko/nameko/pull/784 for more details.
